### PR TITLE
Spoi 10646 upgrade demo apps

### PR DESCRIPTION
**Pull request by francisf**
@tushargosavi Please review
As discussed, following exception was seen
```
Caused by: java.lang.NullPointerException
	at com.datatorrent.netlet.util.Slice.<init>(Slice.java:42)
	at com.datatorrent.contrib.hdht.HDHTReader.<clinit>(HDHTReader.java:250)
	... 11 more
```
Primarily seen due to netlet error (master has fix for null check but not present in tag 1.3.0 which is the latest available in the mvn repository). As discussed have made change in Megh rather than Apex-Core(+netlet version up). 

Since malhar-version was bumped up checkstyle violations for trailing spaces were thrown.

### CODE REVIEW

**Comment by vrozov** [comment]
The fix for NullPointerException is available in netlet 1.3.0 and it is posted to maven central repository.

File: contrib/src/main/java/com/datatorrent/contrib/dimensions/AbstractAppDataDimensionStoreHDHT.java [hunk 1]
```diff
@@ -127,7 +127,7 @@
   private Unifier<String> queryResultUnifier;
 
   protected long responseDelayMillis;
-  
+
```
Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/dimensions/AbstractAppDataDimensionStoreHDHT.java]: All format/codestyle related changes must be done in a separate commit (or even in a separate PR with separate JIRA) using Malhar Jenkins user.

File: contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTReader.java [hunk 2]
```diff
@@ -247,7 +247,7 @@ protected synchronized void invalidateReader(long bucketKey, Set<String> fileNam
     }
   }
 
-  private static Slice GET_KEY = new Slice(null, 0, 0);
+  private static Slice GET_KEY = new Slice(new byte[] {0}, 0, 1);
```
Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTReader.java]: Not needed

File: contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTReader.java [hunk 3]
```diff
@@ -274,7 +274,7 @@ protected synchronized void invalidateReader(long bucketKey, Set<String> fileNam
           bucket.readers.put(floorEntry.getValue().name, reader =
               store.getReader(bucketKey, floorEntry.getValue().name));
         }
-        Slice value = new Slice(null, 0,0);
+        Slice value = new Slice(new byte[] {0}, 0,1);
```
Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTReader.java]: Not needed

File: contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java [hunk 4]
```diff
@@ -465,8 +466,8 @@ private void writeFileWithPurge(Bucket bucket, BucketFileMeta meta, RangeSet<Sli
     TreeMap<Slice, Slice> fileData = new TreeMap<>(keyComparator);
 
     /* Check if there is data in initial part of file before next purge range */
-    Slice key = new Slice(null, 0, 0);
-    Slice value = new Slice(null, 0, 0);
+    Slice key = new Slice(new byte[] {0}, 0, 1);
```
Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java]: Not needed

File: pom.xml [hunk 5]
```diff
@@ -23,7 +23,7 @@
   <parent>
     <groupId>org.apache.apex</groupId>
     <artifactId>malhar</artifactId>
-    <version>3.5.0</version>
+    <version>3.6.0</version>
```
Comment (vrozov) [inline pom.xml]: Why is it necessary to upgrade malhar version? Is there any particular issue fixed in malhar 3.6.0 that justifies the upgrade?

**Review by vrozov — CHANGES_REQUESTED** [review]

File: contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java (hunk 4 above)
Comment (francisf) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java]: @vrozov There seems to be a difference in [tag 1.3.0](https://github.com/DataTorrent/Netlet/blob/v1.3.0/src/main/java/com/datatorrent/netlet/util/Slice.java#L42) and [master](https://github.com/DataTorrent/Netlet/blob/master/src/main/java/com/datatorrent/netlet/util/Slice.java#L42) for the null check. @tushargosavi decompiled netlet 1.3.0 from the maven repo and we see that the null check is missing.

File: contrib/src/main/java/com/datatorrent/contrib/dimensions/AbstractAppDataDimensionStoreHDHT.java (hunk 1 above)
Comment (francisf) [inline contrib/src/main/java/com/datatorrent/contrib/dimensions/AbstractAppDataDimensionStoreHDHT.java]: @vrozov I'll separate the commit in another PR. Can you point me to some docs that describe how  to do this as malhar jenkins user?

Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/dimensions/AbstractAppDataDimensionStoreHDHT.java]: `git commit --author=<jenkins@datatorrent.com>`. Please open another JIRA. Format changes are not directly related to the upgrade.

File: contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java (hunk 4 above)
Comment (vrozov) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/HDHTWriter.java]: tag 1.3.0 was not correctly applied in Datatorrent/Netlet. I fixed the tag.

**Pull request closed without merging**